### PR TITLE
Extract internal logic from lua layer to dedicated files

### PR DESCRIPTION
**Pull request by rofferom**
As a preliminary step to an high-level C API, it is required to isolate as much generic logic as possible from the Lua layer. This extracted code will be used by both the Lua layer and the high-level C API.

This PR is a preliminary draft that shows the parts to extract from the Lua layer:
* `set_epoch()` logic
* Control calls logic
* `commit()`/`discard()` logic
* `link()` logic
* IOSys enumeration/creation

All the file naming and function naming have been done quickly and must be reviewed/improved. However, this is a working base that allowed to crate a prototype of the high-level API.

### CODE REVIEW

File: src/include/libtxproto/epoch.h [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+typedef struct EpochEventCtx {
+    enum EpochMode mode;
+    int64_t value;
+    AVBufferRef *src_ref;
+
+    int (*external_cb)(int64_t systemtime, AVBufferRef *arg);
+    AVBufferRef *external_arg;
+} EpochEventCtx;
+
+AVBufferRef *epoch_event_new(TXMainContext *ctx);
```
Comment (cyanreg) [inline src/include/libtxproto/epoch.h]: Could you prefix all external functions with sp_?

File: src/meson.build [hunk 2]
```diff
@@ -161,7 +171,7 @@ meson.override_dependency('libtxproto', libtxproto)
 # Install headers
 foreach h : headers
     parts = h.split('/')
-    path = meson.project_name()
+    path = 'lib'+meson.project_name()
```
Comment (cyanreg) [inline src/meson.build]: Err, is this necessary?

**Comment by cyanreg** [comment]
Overall, I like this. With those minor changes, I think it'll be good enough to merge and improve.
Thanks

File: src/include/libtxproto/epoch.h (hunk 1 above)
Comment (rofferom) [reply src/include/libtxproto/epoch.h]: Done

Comment (rofferom) [reply src/include/libtxproto/epoch.h]: Done

File: src/meson.build (hunk 2 above)
Comment (rofferom) [reply src/meson.build]: Removed

**Pull request merged**